src/main/java/ch/njol/skript/expressions/ExprTarget.java

```diff
@@ -153,21 +165,43 @@ public String toString(@Nullable Event event, boolean debug) {
 	/**
 	 * Gets an entity's target.
 	 *
-	 * @param entity The entity to get the target of
+	 * @param origin The entity to get the target of.
 	 * @param type The exact EntityData to find. Can be null for any entity.
-	 * @return The entity's target
+	 * @return The entity's target.
 	 */
 	@Nullable
 	@SuppressWarnings("unchecked")
-	public static <T extends Entity> T getTarget(LivingEntity entity, @Nullable EntityData<T> type) {
-		if (entity instanceof Mob)
-			return ((Mob) entity).getTarget() == null || type != null && !type.isInstance(((Mob) entity).getTarget()) ? null : (T) ((Mob) entity).getTarget();
-
-		RayTraceResult result = entity.rayTraceEntities(SkriptConfig.maxTargetBlockDistance.value());
+	public static <T extends Entity> T getTarget(LivingEntity origin, @Nullable EntityData<T> type) {
+		if (origin instanceof Mob)
+			return ((Mob) origin).getTarget() == null || type != null && !type.isInstance(((Mob) origin).getTarget()) ? null : (T) ((Mob) origin).getTarget();
+		Location location = origin.getLocation();
+		RayTraceResult result = null;
+		double raySize = 0.0D;
+		// TODO when DisplayData is added.
+//		if (type.getClass().equals(DisplayData.class))
+//			raySize = 1.0D;
```

Comment:
I don't think this is appropriate, the target entity should reflect the actual target of the player based on hitboxes. If the player wants to expand hitboxes for displays, they should do so themselves via an exposed option for raysize.

Reply:
I agree they'll probably assume that, but simply expanding the hitbox isn't a good way of achieving that, since we don't know how big the display is, if it's rotated, translated, whatever. Simply expanding by 1 leaves tons of room for false positives. 

You're right, though, we can discuss this further when it's actually added